refactor: clean up clip image encoder (#213)
jina-ai/executors: jinahub/encoders/image/CLIPImageEncoder/README.md

# CLIPImageEncoder

**CLIPImageEncoder** is an image encoder that wraps the image embedding functionality using the [CLIP](https://huggingface.co/transformers/model_doc/clip.html) model from huggingface transformers.

> This encoder is meant to be used in conjunction with the [CLIPTextEncoder](https://github.com/jina-ai/executors/tree/main/jinahub/encoders/text/CLIPTextEncoder),

Reviewer: Use the link to the jina hub
```suggestion
This encoder is meant to be used in conjunction with the [CLIPTextEncoder](https://hub.jina.ai/executor/livtkbkg),

```

> as it can embedd text and images to the same latent space.

Reviewer: ```suggestion
as it can embed text and images to the same latent space.

```

## Usage

Use the prebuilt images from JinaHub in your Python code,

```python
from jina import Flow, Document

f = Flow().add(
    uses='jinahub+docker://CLIPImageEncoder',
)
```

or in the .yml config.

```yaml
jtype: Flow
pods:
  - name: encoder

>     uses: 'jinahub+docker://AudioCLIPTextEncoder'

Reviewer: ```suggestion
    uses: 'jinahub+docker://CLIPImageEncoder'

```

Author: good catch!

# CLIPImageEncoder

**CLIPImageEncoder** is an image encoder that wraps the image embedding functionality using the [CLIP](https://huggingface.co/transformers/model_doc/clip.html) model from huggingface transformers.

This encoder is meant to be used in conjunction with the [CLIPTextEncoder](https://github.com/jina-ai/executors/tree/main/jinahub/encoders/text/CLIPTextEncoder),
as it can embedd text and images to the same latent space.

The following arguments can be passed on initialization:

> - `pretrained_model_name_or_path`: name or path of the pre-trained CLIP model.

Reviewer 2: We don't want the arguments in the README, right? They are parsed by Hubble and displayed in the hub.jina.ai interface

- `base_feature_extractor`: Base feature extractor for images. Defaults to ``pretrained_model_name_or_path`` if None.
- `use_default_preprocessing`: Whether to use the `base_feature_extractor` on images (blobs) before encoding them, default is `True`.
- `device`: Pytorch device to put the model on, e.g. 'cpu', 'cuda'.
- `traversal_paths`: traversal path (use `cpu` if not specified in request's parameters).
- `batch_size`: default batch size (use `32` if not specified in request's parameters).

> #### Inputs 

Reviewer 2: Same for inputs. Check https://github.com/jina-ai/executor-simpleindexer

# CLIPImageEncoder

**CLIPImageEncoder** is an image encoder that wraps the image embedding functionality using the [CLIP](https://huggingface.co/transformers/model_doc/clip.html) model from huggingface transformers.
This encoder is meant to be used in conjunction with the [CLIPTextEncoder](https://hub.jina.ai/executor/livtkbkg),
as it can embed text and images to the same latent space.

Use the prebuilt images from JinaHub in your Python code,

```python
from jina import Flow

> f = Flow().add(uses='jinahub+docker://CLIPImageEncoder',)

Reviewer 2: same as in the other PR. Again, look at https://hub.jina.ai/executor/zb38xlt
![Screenshot from 2021-09-17 16-10-06](https://user-images.githubusercontent.com/8330330/133796753-c14108fb-9255-4b81-a58b-0be6ea9d418e.png)
4
